Platform/OpenNetworkBoot/README.md

```diff
@@ -0,0 +1,396 @@
+
+# Run as PXE Boot DHCP proxy for specified network (use default /24 network size)
+dhcp-range=192.168.2.0,proxy
+
+# Trigger PXE Boot support on HTTP Boot client request
+dhcp-pxe-vendor=HTTPClient
+
+# Set triggering tag if correct arch is present in option 60
+dhcp-match=set:arch_x64,option:client-arch,16
+
+# Make PXE Boot support believe it has something to send...
+pxe-service=tag:arch_x64,x86-64_EFI,"Network Boot"
+
+# Specify bootfile-name via PXE Boot setting
+dhcp-boot=tag:arch_x64,https://michaels-air.lan:8443/images/OpenShell.efi
+
+# Force required vendor class in response, even if not requested
+dhcp-option-force=tag:arch_x64,option:vendor-class,HTTPClient
+```
+
+An HTTP server (such as Apache, nginx, or multiple other options) will be
+required to serve the actual NBP files over `http://` or `https://`.
+
+References:
+ - https://github.com/ipxe/ipxe/discussions/569
+ - https://www.mail-archive.com/dnsmasq-discuss@lists.thekelleys.org.uk/msg16278.html
+
+### HTTPS Boot
+
+Note that the certificate for validating `https://` requests should be loaded into
+firmware using the OpenNetworkBoot `--enroll-cert` option.
+
+A normal https site would not serve files using a self-signed certificate
+authority (CA), but since we are only attempting to serve files to HTTP boot
+clients, in this case we can do so.
+
+## Booting ISO and IMG files
+
+Though not often noted in the documentation, the majority of HTTP Boot
+implementations support loading `.iso` and `.img` files, which will be
+automatically mounted as a ramdisk. If the mounted filesystem includes
+`\EFI\BOOT\BOOTx64.efi` (or `\EFI\BOOT\BOOTIA32.efi` for 32-bit) then this
+file will be loaded from the ramdisk and started.
+
+The MIME types corresponding to `.iso` and `.img` files are:
+
+ - `application/vnd.efi-iso`
+ - `application/vnd.efi-img`
+
+The MIME type for `.efi` files is:
+
+ - `application/efi`
+
+If the MIME type is none of the above, then the corresponding file
+extensions (`.iso`, `.img` and `.efi`) are used instead to identify the NBP
+file type.
+
+> **Note**: Files which cannot be recognised by any of the above MIME types or
+file extensions will _not_ be loaded by standard HTTP boot drivers (including
+the `HttpBootDxe` driver currently shipped with OpenCore).
+
+## Booting DMG files
+
+In order to allow booting macOS Recovery, `OpenNetworkBoot` includes
+additional support for loading `.dmg` files via HTTP boot. If the NBP
+filename is `{filename}.dmg` or `{filename}.chunklist` then the other
+file of this pair will be automatically loaded, in order to allow DMG
+chunklist verification, and both files will be used for OpenCore DMG booting.
+
+### MIME types
+
+In order for `.dmg` and `.chunklist` files to be loaded by standard HTTP boot
+drivers (including the `HttpBootDxe` driver currently shipped with OpenCore),
+they must be served with the MIME type `application/efi`, otherwise they will
+be treated as having an unknown file type and will not be loaded.
```

Comment:
If we are creating a custom OpenHttpBootDxe (i.e. to support a new protocol, allowing loading DMGs directly into the required OC-format DMG ramdisk), then at the same time we might as well implement support for recognising `.dmg` and `.chunklist` files directly (rather than requiring users to faff around setting not-really-correct MIME types to serve them).

Reply:
It's occurred to me that since I already trap an HTTP callback, it may be possible to forcibly fake the MIME type that HttpBootDxe sees for these files. In which case these files could be recognised by extension, and no special MIME type would need to be set on the web server, even for standard HttpBootDxe. I will look into this...

Reply:
Have implemented via callback, 'fake' MIME type no longer needed.